packages/api/src/jobs/certification/record_d_generation.js

```diff
@@ -26,119 +27,160 @@ const dataTransformsMapping = {
   treated: treatedTransformation,
   treated_doc: blankTransformation,
   genetically_engineered: blankTransformation,
-}
+};
 
 module.exports = (data, exportId, from_date, to_date, farm_name) => {
-  return XlsxPopulate.fromBlankAsync()
-    .then((workbook) => {
-      const defaultStyles = {
-        verticalAlignment: 'center',
-        fontFamily: 'Calibri',
-        fill: 'F2F2F2',
-      }
-      const RichText = XlsxPopulate.RichText;
-      const rowSix = new RichText();
-      const rowSeven = new RichText();
-      const rowEight = new RichText();
-      const rowNine = new RichText();
-      const reportDate = new Date().toISOString().split('T')[0];
+  return XlsxPopulate.fromBlankAsync().then((workbook) => {
+    const defaultStyles = {
+      verticalAlignment: 'center',
+      fontFamily: 'Calibri',
+      fill: 'F2F2F2',
+    };
+    const RichText = XlsxPopulate.RichText;
+    const rowSix = new RichText();
+    const rowSeven = new RichText();
+    const rowEight = new RichText();
+    const rowNine = new RichText();
+    const reportDate = new Date().toISOString().split('T')[0];
 
-      const { t } = i18n;
-      rowSix.add(`2. ${t('RECORD_D.NOTE.TWO.PART_1')} `)
-        .add(` ${t('RECORD_D.NOTE.TWO.PART_2')} `, { bold: true })
-        .add(`${t('RECORD_D.NOTE.TWO.PART_3')}:`);
-      rowSeven.add(`     A.  ${t('RECORD_D.NOTE.TWO.A')}`);
-      rowEight.add(`     B.  ${t('RECORD_D.NOTE.TWO.B')}`);
-      rowNine.add(`     C.  ${t('RECORD_D.NOTE.TWO.C')}`);
-      workbook.sheet(0).cell('A1').value(t('RECORD_D.HEADER'));
-      workbook.sheet(0).range('A1:I1').style({
-        bold: true,
-        fontSize: 20,
-        fill: 'F2F2F2',
-      });
-      workbook.sheet(0).range('A6:I9').style({
-        fill: 'F2F2F2',
-      });
+    rowSix
+      .add(`2. ${t('RECORD_D.NOTE.TWO.PART_1')} `)
```

Comment:
Just wanted to be sure you knew about the spreadsheet library limitation: trailing spaces like the one here may or may not be preserved, depending on the specific spreadsheet software used to view the result. The library does the proper xml encoding on *leading* spaces so that they are always preserved. Not sure if that matters here.

Reply:
Fixed